go/buildhelper/lib.go

```diff
@@ -0,0 +1,43 @@
+package buildhelper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+var CONNECTION_TIMEOUT = 15 * time.Second
+
+// EthAPI struct
+// based on https://eth.wiki/json-rpc/API
+type EthAPI struct {
+	ApiClient *ethclient.Client
+	retry     int
+	timeout   time.Duration
+	ipaddress string
+	port      uint
+	ctx       context.Context
+}
+
+func NewEthAPI(ipaddress string, port uint) *EthAPI {
+	return &EthAPI{
+		timeout:   CONNECTION_TIMEOUT,
+		ipaddress: ipaddress,
+		port:      port,
+		ctx:       context.Background(),
+	}
+}
+
+func (e *EthAPI) Connect() error {
+	var err error
+	for start := time.Now(); time.Since(start) < CONNECTION_TIMEOUT; time.Sleep(time.Second) {
```

Comment:
@otherview Is this loop the right approach? Even if `Dial` returns a meaningful error (e.g. IP address is invalid), you'll be slamming the client for 15 seconds. If it's transitory (maybe the server isn't listening yet), I'd put a little sleep in between dials at least (e.g. 100 millis)

Reply:
It's sleeping 1 second between attempts :)